pkg/tenant/members.go

```diff
@@ -10,8 +10,39 @@ import (
 	"github.com/rs/zerolog/log"
 )
 
+// TenantMemberList retrieves all members assigned to a tenant
+// and returns a 200 OK response.
+//
+// Route: tenant/:tenantID/member
 func (s *Server) TenantMemberList(c *gin.Context) {
-	c.JSON(http.StatusNotImplemented, "not implemented yet")
+	var (
+		err error
+	)
+
+	// Get the member's tenant ID from the URL and return a 400 response
+	// if the tenant ID is not a ULID.
+	var tenantID ulid.ULID
+	if tenantID, err = ulid.Parse(c.Param("tenantID")); err != nil {
+		log.Error().Err(err).Msg("could not parse tenant ulid")
+		c.JSON(http.StatusBadRequest, api.ErrorResponse("could not parse tenant ulid"))
+		return
+	}
+
+	// Get members from the database and return a 500 response
+	// if not successful.
+	if _, err := db.ListMembers(c.Request.Context(), tenantID); err != nil {
+		log.Error().Err(err).Msg("could not fetch members from the database")
+		c.JSON(http.StatusInternalServerError, api.ErrorResponse("could not fetch members from the database"))
+		return
+	}
+
+	out := &api.TenantMemberPage{TenantMembers: make([]*api.Member, 0)}
+
+	tenantMember := &api.Member{}
+
+	out.TenantMembers = append(out.TenantMembers, tenantMember)
```

Comment:
It looks like on line 33 we have the potential to obtain a slice of members (type `[]*Member`) from the database, but we're currently assigning the result to a `_`. Should we instead be capturing the result into a new var such as `members` and then assigning `out.TenantMembers = members` rather than appending an empty `tenantMember`?

Reply:
Line 33 obtains a slice of `[]*db.Member` and this cannot be assigned to `out.TenantMembers`. I've attempted a few different things, following your suggestion, but can't quite get anything to work along those lines. I'll give this another try on Monday and I haven't made any progress.